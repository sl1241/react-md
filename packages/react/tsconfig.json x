{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
