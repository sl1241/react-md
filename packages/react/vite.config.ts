import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
