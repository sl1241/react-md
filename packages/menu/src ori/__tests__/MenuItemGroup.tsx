import { render } from "@testing-library/react";

import { MenuItemGroup } from "../MenuItemGroup";

describe("MenuItemGroup", () => {
  it("should require an aria-label or aria-labelledby for a11y", () => {
    const props = { children: <li>Content</li> };

    // @ts-expect-error
    const { container, rerender } = render(<MenuItemGroup {...props} />);
    expect(container).toMatchSnapshot();

    rerender(<MenuItemGroup aria-label="Group" {...props} />);
    expect(container).toMatchSnapshot();

    rerender(<MenuItemGroup aria-labelledby="non-existing-id" {...props} />);
    expect(container).toMatchSnapshot();
  });
});
