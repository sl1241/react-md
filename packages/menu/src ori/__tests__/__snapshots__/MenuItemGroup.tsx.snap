// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemGroup should require an aria-label or aria-labelledby for a11y 1`] = `
<div>
  <ul
    class="rmd-list"
    role="group"
  >
    <li>
      Content
    </li>
  </ul>
</div>
`;

exports[`MenuItemGroup should require an aria-label or aria-labelledby for a11y 2`] = `
<div>
  <ul
    aria-label="Group"
    class="rmd-list"
    role="group"
  >
    <li>
      Content
    </li>
  </ul>
</div>
`;

exports[`MenuItemGroup should require an aria-label or aria-labelledby for a11y 3`] = `
<div>
  <ul
    aria-labelledby="non-existing-id"
    class="rmd-list"
    role="group"
  >
    <li>
      Content
    </li>
  </ul>
</div>
`;
